buildSrc/src/main/java/org/elasticsearch/gradle/test/rest/CopyRestApiPlugin.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.gradle.test.rest;
+
+import org.elasticsearch.gradle.VersionProperties;
+import org.elasticsearch.gradle.info.BuildParams;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.artifacts.Dependency;
+import org.gradle.api.provider.Provider;
+
+import java.util.Map;
+
+/**
+ * Gradle plugin to help configure {@link CopyRestApiTask}'s that will copy the artifacts needed for the Rest API spec and YAML tests.
+ * @see CopyRestApiTask
+ */
+public class CopyRestApiPlugin implements Plugin<Project> {
+
+    private static final String COPY_SPEC_EXTENSION_NAME = "copyRestApiSpecs";
+    private static final String COPY_TEST_EXTENSION_NAME = "copyYamlTests";
+
+    @Override
+    public void apply(Project project) {
+        CopyRestApiExtension copySpecExtension = project.getExtensions().create(COPY_SPEC_EXTENSION_NAME, CopyRestApiExtension.class);
```

Comment:
We should avoid creating two extensions here. For a couple of reasons:
1. It's not clear from the model these two are related or "owned" by a single plugin. Having a single extension encourages putting this stuff together.
1. Because we are using the same _type_ for this extension, it then means folks can't effectively use `extensions.getByType()` since there would be ambiguity here.

I suggest we do the following:
1. Rename `CopyRestApiExtension` to `RestResourcesSpec`.
1. Create a new `RestResourcesExtension` with two fields, `restApi` and `restTests`.
1. `RestResourcesExtension` should have methods to configure those fields that look like `void restApi(Action<? super RestResourcesSpec> spec)`.
1. Add the single `RestResourcesExtension` to the project.

We should then have a DSL looking like this:

```
restResources {
  restApi {
    includeXpack 'ilm', 'slm'
  }
  restTests {
    includeXpack 'ilm', 'slm'
  }
}
```

Reply:
FYI, i use the term "rest resources" here because I think it's the best generic term. Please suggest a better one. I hesitate to use the term "rest testing" or similar due to your earlier comment about api specs being used for purposes not necessarily related to testing.

Reply:
I like that, and thanks for the detailed steps!